### PR TITLE
add nf-test for star/genomegenerate

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/star/genomegenerate/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert path(process.out.index.get(0).get(1)).exists() },
+                { assert path(process.out.versions.get(0)).getText().contains("star") },
+                { assert path(process.out.versions.get(0)).getText().contains("samtools") },
+                { assert path(process.out.versions.get(0)).getText().contains("gawk") }
+            )
```
Comment (mashehu) [inline modules/nf-core/star/genomegenerate/tests/main.nf.test:35]: 👀 🙂

Comment (maxulysse) [reply modules/nf-core/star/genomegenerate/tests/main.nf.test:35]: Yeah, I know, I asked yesterday already, snapshot is stable but not working on conda

Comment (mashehu) [reply modules/nf-core/star/genomegenerate/tests/main.nf.test:35]: ah, it's for this one I see. Sorry, didn't recognize it. Do you mind investigating this a bit more? Are there maybe different version of star etc. in conda compared to the other containers?

Comment (mashehu) [reply modules/nf-core/star/genomegenerate/tests/main.nf.test:35]: okay, had a look and found I prime example of why to snapshot the version:

`environment.yml`:
```yaml
 - bioconda::star=2.7.10a
 ```

container:
```
'star=2.7.9a'
```

Comment (maxulysse) [reply modules/nf-core/star/genomegenerate/tests/main.nf.test:35]: Looking forward to wave

File: modules/nf-core/star/genomegenerate/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                    [file(params.test_data['homo_sapiens']['genome']['genome_gtf'], checkIfExists: true)]
+                ])
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert path(process.out.index.get(0).get(1)).exists() },
```
Comment (nvnieuwk) [inline modules/nf-core/star/genomegenerate/tests/main.nf.test:31]: I personally like to snapshot the name of files that I want to check the existence of. That way you also check that the output naming stays correct. (Just a suggestion, this is definitely fine too)

Comment (nvnieuwk) [reply modules/nf-core/star/genomegenerate/tests/main.nf.test:31]: `file(process.out.index[0][1]).name` would do that BTW

File: modules/nf-core/star/genomegenerate/tests/main.nf.test (hunk 1 above)
Comment (maxulysse) [reply modules/nf-core/star/genomegenerate/tests/main.nf.test:35]: ok, so @FriederikeHanssen is the one that created the last containers.
and she did request the right version.
cf https://github.com/BioContainers/multi-package-containers/pull/2377/files#diff-f6be21f33f5bb216e9a0baac781bc0d81da47c65d70a53701dfc6cac35c80966R263

so something is wrong on the biocontainers side

Comment (FriederikeHanssen) [reply modules/nf-core/star/genomegenerate/tests/main.nf.test:35]: I updated both actually with star .10 and .9 and must have used the wrong tag to update :(

Comment (maxulysse) [reply modules/nf-core/star/genomegenerate/tests/main.nf.test]: Yeah, that was probably it.
thanks for your help anyway

**Review by mashehu — APPROVED** [review]

**Review by nvnieuwk — APPROVED** [review]

**Pull request merged**